platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/locklistener/RejectingFileLockContentionHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal.locklistener;
+
+import org.gradle.cache.FileLockReleasedSignal;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+import java.util.function.Consumer;
+
+/**
+ * A special contention handler that doesn't support communicating contention.
+ */
+@NullMarked
+public class RejectingFileLockContentionHandler implements FileLockContentionHandler {
```

Comment:
❓ What happens if we use `OnDemandEagerRelease` in Gradle? As far as I read code there we will always use `DefaultFileLockContentionHandler`, right. Is there any harm in that?

Should we maybe document in `OnDemandEagerRelease` JavaDoc that for Gradle we should use ``OnDemand` or alternatively should we choose the right `FileLockContentionHandler`in `DefaultFileLockManager` by some `LockOptions` flag?